cpp/new_journal_alert.cc

```diff
@@ -172,6 +206,220 @@ namespace std {
+                                       const std::vector<NewIssueInfo> &new_issue_infos) const
+     {
+         std::string email_contents("Dear " + name_of_user + ",<br /><br />\n"
+                                    "An automated process has determined that new issues are available for\n"
+                                    "serials that you are subscribed to.  The list is:\n"
+                                    "<ul>\n"); // start journal list
+
+         std::string last_series_title, last_volume_year_and_issue;
+         for (const auto &new_issue_info : new_issue_infos) {
+             const bool new_serial(new_issue_info.series_title_ != last_series_title);
+             if (new_serial) {
+                 if (not last_series_title.empty()) { // Not first iteration!
+                     email_contents += "    </ul>\n"; // end items
+                     email_contents += "  </ul>\n"; // end volume/year/issue list
+                 }
+                 last_series_title = new_issue_info.series_title_;
+                 email_contents += "  <li>" + HtmlUtil::HtmlEscape(last_series_title) + "</li>\n";
+                 email_contents += "  <ul>\n"; // start volume/year/issue list
+                 last_volume_year_and_issue.clear();
+             }
+
+             // Generate "volume_year_and_issue":
+             std::string volume_year_and_issue;
+             if (not new_issue_info.volume_.empty())
+                 volume_year_and_issue += new_issue_info.volume_;
+             if (not new_issue_info.year_.empty()) {
+                 if (not volume_year_and_issue.empty())
+                     volume_year_and_issue += " ";
+                 volume_year_and_issue += "(" + new_issue_info.year_ + ")";
+             }
+             if (not new_issue_info.issue_.empty()) {
+                 if (not volume_year_and_issue.empty())
+                     volume_year_and_issue += ", ";
+                 volume_year_and_issue += new_issue_info.issue_;
+             }
+
+             if (volume_year_and_issue != last_volume_year_and_issue) {
+                 if (not new_serial)
+                     email_contents += "    </ul>\n"; // end items
+                 email_contents += "    <li>" + HtmlUtil::HtmlEscape(volume_year_and_issue) + "</li>\n";
+                 last_volume_year_and_issue = volume_year_and_issue;
+                 email_contents += "    <ul>\n"; // start items
+             }
+
+             const std::string URL("https://" + vufind_host + "/Record/" + new_issue_info.control_number_);
+             std::string authors;
+             for (const auto &author : new_issue_info.authors_)
+                 authors += "&nbsp;&nbsp;&nbsp;" + HtmlUtil::HtmlEscape(author);
+             email_contents += "      <li><a href=" + URL + ">" + HtmlUtil::HtmlEscape(new_issue_info.issue_title_) + "</a>" + authors
```

Comment:
Dont forget to export double quotes around the value of the href attribute.

Reply:
Done.